python/social_bot/tasks.py

```diff
@@ -460,8 +480,13 @@ def _move_goal_impl(self, goal, agent_loc, agent_dir):
                        random.random() * range - range / 2, 0)
 
             self._initial_dist = np.linalg.norm(loc - agent_loc)
-            if self._initial_dist > self._success_distance_thresh:
+            if self._initial_dist > self._success_distance_thresh and (
+                attempts > 10000 or (
```

Comment:
In what case will the attempts > 10000 and still can not found a valid position? Maybe log.e() some information if attempts > 10000 ?